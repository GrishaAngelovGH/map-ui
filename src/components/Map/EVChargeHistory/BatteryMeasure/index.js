import BatteryMeasure from './BatteryMeasure'

export default BatteryMeasure