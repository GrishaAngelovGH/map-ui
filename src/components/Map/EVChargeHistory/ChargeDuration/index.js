import ChargeDuration from './ChargeDuration'

export default ChargeDuration