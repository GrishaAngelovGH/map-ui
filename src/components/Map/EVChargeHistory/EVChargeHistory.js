import Modal from 'react-bootstrap/Modal'

const EVChargeHistory = ({ show, onHide }) => {
  return (
    <Modal show={show} onHide={onHide}>
      <Modal.Header closeButton>
        <Modal.Title>
          <i className='bi bi-ev-station'></i>
          <span className='mx-2'>EV Charge History</span>
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
      </Modal.Body>
    </Modal>
  )
}

export default EVChargeHistory