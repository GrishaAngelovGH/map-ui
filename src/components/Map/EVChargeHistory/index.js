import EVChargeHistory from './EVChargeHistory'

export default EVChargeHistory