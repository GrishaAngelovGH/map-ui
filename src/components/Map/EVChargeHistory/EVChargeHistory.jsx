import Modal from 'react-bootstrap/Modal'
import BatteryMeasure from './BatteryMeasure'
import ChargeDuration from './ChargeDuration'
import Summary from './Summary'

const EVChargeHistory = ({ show, onHide }) => {
  const history = [
    { id: 1, location: { city: 'London', country: 'England, UK' }, battery: 100, charge: false, distanceInKm: 20, durationTimeInMin: 30 },
    { id: 2, location: { city: 'London', country: 'England, UK' }, battery: 90, charge: false, distanceInKm: 24, durationTimeInMin: 28 },
    { id: 3, location: { city: 'London', country: 'England, UK' }, battery: 70, charge: false, distanceInKm: 46, durationTimeInMin: 55 },
    { id: 4, location: { city: 'London', country: 'England, UK' }, currentBatteryCharge: 85, prevBatteryCharge: 70, durationTimeInMin: 35, charge: true, name: 'ChargePoint - 123 Road', address: 'A12 B34 London, UK' },
    { id: 5, location: { city: 'London', country: 'England, UK' }, battery: 65, charge: false, distanceInKm: 32, durationTimeInMin: 40 },
    { id: 6, location: { city: 'London', country: 'England, UK' }, battery: 60, charge: false, distanceInKm: 5, durationTimeInMin: 10 },
    { id: 7, location: { city: 'London', country: 'England, UK' }, currentBatteryCharge: 95, prevBatteryCharge: 60, durationTimeInMin: 75, charge: true, name: 'ChargePoint - 123 Road', address: 'A12 B34 London, UK' }
  ]

  return (
    <Modal show={show} onHide={onHide}>
      <Modal.Header closeButton>
        <Modal.Title>
          <i className='bi bi-ev-station'></i>
          <span className='mx-2'>EV Charge History</span>
        </Modal.Title>
      </Modal.Header>
      <Modal.Body className='overflow-auto' style={{ height: 500 }}>
        <Summary history={history} />

        <div className='p-3'>
          {
            history.map(v => (
              v.charge ?
                <ChargeDuration key={v.id} duration={v} /> :
                <BatteryMeasure key={v.id} measure={v} />
            ))
          }
        </div>
      </Modal.Body>
    </Modal>
  )
}

export default EVChargeHistory