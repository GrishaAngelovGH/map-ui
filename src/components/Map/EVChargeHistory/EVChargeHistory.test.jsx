import { render } from '@testing-library/react'

import EVChargeHistory from './EVChargeHistory'

test('should render EVChargeHistory component', () => {
  const view = render(<EVChargeHistory show={true} />)

  expect(view).toMatchSnapshot()
})