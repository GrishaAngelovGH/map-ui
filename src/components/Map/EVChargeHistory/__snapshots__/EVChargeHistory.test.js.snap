// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EVChargeHistory component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              <i
                class="bi bi-ev-station"
              />
              <span
                class="mx-2"
              >
                EV Charge History
              </span>
            </div>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="overflow-auto modal-body"
            style="height: 500px;"
          >
            <div
              class="row justify-content-center text-center"
            >
              <div
                class="col-md-3"
              >
                <p
                  class="m-0 text-secondary"
                >
                  Distance
                </p>
                <p
                  class="fw-bold"
                >
                  223 km
                </p>
              </div>
              <div
                class="col-md-4"
              >
                <p
                  class="m-0 text-secondary"
                >
                  Total Duration
                </p>
                <p
                  class="fw-bold"
                >
                  4h 52m
                </p>
              </div>
              <div
                class="col-md-3"
              >
                <p
                  class="m-0 text-secondary"
                >
                  Charge
                </p>
                <p
                  class="fw-bold"
                >
                  56m
                </p>
              </div>
            </div>
            <div
              class="p-3"
            >
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-2"
                >
                  <i
                    class="bi bi-car-front fs-1"
                  />
                </div>
                <div
                  class="col-md-7"
                >
                  <p
                    class="m-0 fw-bold"
                  >
                    London
                  </p>
                  <p
                    class="m-0 text-secondary"
                  >
                    London
                    , 
                    England, UK
                  </p>
                </div>
                <div
                  class="col-md-3 d-flex align-items-center"
                >
                  <i
                    class="bi bi-battery-full fs-1"
                  />
                  <span
                    class="fw-bold mx-2"
                  >
                    100
                    %
                  </span>
                </div>
              </div>
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-2"
                >
                  <i
                    class="bi bi-car-front fs-1"
                  />
                </div>
                <div
                  class="col-md-7"
                >
                  <p
                    class="m-0 fw-bold"
                  >
                    London
                  </p>
                  <p
                    class="m-0 text-secondary"
                  >
                    London
                    , 
                    England, UK
                  </p>
                </div>
                <div
                  class="col-md-3 d-flex align-items-center"
                >
                  <i
                    class="bi bi-battery-half fs-1"
                  />
                  <span
                    class="fw-bold mx-2"
                  >
                    90
                    %
                  </span>
                </div>
              </div>
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-2"
                >
                  <i
                    class="bi bi-car-front fs-1"
                  />
                </div>
                <div
                  class="col-md-7"
                >
                  <p
                    class="m-0 fw-bold"
                  >
                    London
                  </p>
                  <p
                    class="m-0 text-secondary"
                  >
                    London
                    , 
                    England, UK
                  </p>
                </div>
                <div
                  class="col-md-3 d-flex align-items-center"
                >
                  <i
                    class="bi bi-battery-half fs-1"
                  />
                  <span
                    class="fw-bold mx-2"
                  >
                    70
                    %
                  </span>
                </div>
              </div>
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-2"
                    >
                      <i
                        class="bi bi-lightning-charge fs-1"
                      />
                    </div>
                    <div
                      class="col-md-7"
                    >
                      <p
                        class="m-0 fw-bold"
                      >
                        ChargePoint - 123 Road
                      </p>
                      <p
                        class="m-0 text-secondary"
                      >
                        A12 B34 London, UK
                      </p>
                    </div>
                    <div
                      class="col-md-3 d-flex align-items-center"
                    >
                      <i
                        class="bi bi-clock fs-2"
                      />
                      <span
                        class="fw-bold mx-1"
                      >
                        35
                        m
                      </span>
                    </div>
                  </div>
                  <div
                    class="row justify-content-center"
                  >
                    <div
                      class="col-md-4 d-flex align-items-center"
                    >
                      <i
                        class="bi bi-battery-half text-danger fs-1"
                      />
                      <span
                        class="fw-bold text-danger mx-1"
                      >
                        70
                        %
                      </span>
                      <i
                        class="bi bi-arrow-right-short fs-1"
                      />
                    </div>
                    <div
                      class="col-md-4 d-flex align-items-center"
                    >
                      <i
                        class="bi bi-battery-half text-success fs-1"
                      />
                      <span
                        class="fw-bold text-success mx-1"
                      >
                        85
                        %
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-2"
                >
                  <i
                    class="bi bi-car-front fs-1"
                  />
                </div>
                <div
                  class="col-md-7"
                >
                  <p
                    class="m-0 fw-bold"
                  >
                    London
                  </p>
                  <p
                    class="m-0 text-secondary"
                  >
                    London
                    , 
                    England, UK
                  </p>
                </div>
                <div
                  class="col-md-3 d-flex align-items-center"
                >
                  <i
                    class="bi bi-battery-half fs-1"
                  />
                  <span
                    class="fw-bold mx-2"
                  >
                    65
                    %
                  </span>
                </div>
              </div>
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-2"
                >
                  <i
                    class="bi bi-car-front fs-1"
                  />
                </div>
                <div
                  class="col-md-7"
                >
                  <p
                    class="m-0 fw-bold"
                  >
                    London
                  </p>
                  <p
                    class="m-0 text-secondary"
                  >
                    London
                    , 
                    England, UK
                  </p>
                </div>
                <div
                  class="col-md-3 d-flex align-items-center"
                >
                  <i
                    class="bi bi-battery-half fs-1"
                  />
                  <span
                    class="fw-bold mx-2"
                  >
                    60
                    %
                  </span>
                </div>
              </div>
              <div
                class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-2"
                    >
                      <i
                        class="bi bi-lightning-charge fs-1"
                      />
                    </div>
                    <div
                      class="col-md-7"
                    >
                      <p
                        class="m-0 fw-bold"
                      >
                        ChargePoint - 123 Road
                      </p>
                      <p
                        class="m-0 text-secondary"
                      >
                        A12 B34 London, UK
                      </p>
                    </div>
                    <div
                      class="col-md-3 d-flex align-items-center"
                    >
                      <i
                        class="bi bi-clock fs-2"
                      />
                      <span
                        class="fw-bold mx-1"
                      >
                        75
                        m
                      </span>
                    </div>
                  </div>
                  <div
                    class="row justify-content-center"
                  >
                    <div
                      class="col-md-4 d-flex align-items-center"
                    >
                      <i
                        class="bi bi-battery-half text-danger fs-1"
                      />
                      <span
                        class="fw-bold text-danger mx-1"
                      >
                        60
                        %
                      </span>
                      <i
                        class="bi bi-arrow-right-short fs-1"
                      />
                    </div>
                    <div
                      class="col-md-4 d-flex align-items-center"
                    >
                      <i
                        class="bi bi-battery-half text-success fs-1"
                      />
                      <span
                        class="fw-bold text-success mx-1"
                      >
                        95
                        %
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
